idaes/models_extra/power_generation/costing/power_plant_capcost.py

```diff
@@ -892,23 +892,36 @@ def get_PP_costing(
                                 )
                             )
             else:
-                if scaled_param.get_units() is None:
-                    raise ValueError(
-                        "Account %s uses units of %s. "
-                        "Units of %s were passed. "
-                        "Scaled_param must have units."
-                        % (cost_accounts[0], ref_units, scaled_param.get_units())
-                    )
-                else:
-                    try:
-                        pyunits.convert(scaled_param, ref_units)
-                    except InconsistentUnitsError:
-                        raise Exception(
+                try:
+                    if pyunits.get_units(scaled_param) is None:
+                        raise ValueError(
                             "Account %s uses units of %s. "
                             "Units of %s were passed. "
-                            "Cannot convert unit containers."
-                            % (cost_accounts[0], ref_units, scaled_param.get_units())
+                            "Scaled_param must have units."
+                            % (
+                                cost_accounts[0],
+                                ref_units,
+                                pyunits.get_units(scaled_param),
+                            )
                         )
+                    else:
+                        try:
+                            pyunits.convert(scaled_param, ref_units)
+                        except InconsistentUnitsError:
+                            raise Exception(
+                                "Account %s uses units of %s. "
+                                "Units of %s were passed. "
+                                "Cannot convert unit containers."
+                                % (
+                                    cost_accounts[0],
+                                    ref_units,
+                                    pyunits.get_units(scaled_param),
+                                )
+                            )
```

Comment:
Might as well `raise` an `InconsistentUnitsError` here instead of a basic `Exception`.

Reply:
Good call.  I updated the expression types.  Hopfully there aren't specific tests for the exception type.  I guess I'll know soon.